src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -289,7 +302,19 @@ public void resetComments(String project, String repository, String pullRequestI
         
         // delete comment if published by the current SQ user
         if (sonarUser.getId() == comment.getAuthor().getId()) {
-          stashClient.deletePullRequestComment(project, repository, pullRequestId, comment);
+          
+          // comment contains tasks which cannot be deleted => do nothing
+          if (comment.containsNotDeletableTasks()) {
```

Comment:
Ambiguity probably comes from wrong naming of the method "containsNotDeletableTasks".
This method will be renamed as "containsPermanentTasks" to make more readable this code.